prepare_mkl.sh

```diff
@@ -62,26 +62,36 @@ echo $VERSION_LINE  # Return Version Line
 }
 
 # MKL
-DST=`dirname $0`
-OMP=0
-VERSION_MATCH=20120601
-ARCHIVE_BASENAME=mklml_lnx_2017.0.1.20161005.tgz
+HOME_MKL=$2
+if [ ! -d "$HOME_MKL" ]; then
+   mkdir $HOME_MKL
+fi
+MXNET_ROOT=`dirname $0`
+OMP=0 
+VERSION_MATCH=20161123
+ARCHIVE_BASENAME=mklml_lnx_2017.0.2.20161122.tgz
 MKL_CONTENT_DIR=`echo $ARCHIVE_BASENAME | rev | cut -d "." -f 2- | rev`
 MKLURL="https://github.com/dmlc/web-data/raw/master/mxnet/mklml-release/$ARCHIVE_BASENAME"
 # there are diffrent MKL lib to be used for GCC and for ICC
 reg='^[0-9]+$'
 VERSION_LINE=`GetVersionName $MKLROOT`
+#echo $VERSION_LINE
 # Check if MKLROOT is set if positive then set one will be used..
 if [ -z $MKLROOT ] || [ $VERSION_LINE -lt $VERSION_MATCH ]; then
-	# ..if MKLROOT is not set then check if we have MKL downloaded in proper version
-    VERSION_LINE=`GetVersionName $DST/$MKL_CONTENT_DIR`
+  # ..if MKLROOT is not set then check if we have MKL downloaded in proper version
+    VERSION_LINE=`GetVersionName $HOME_MKL`
+    #echo $VERSION_LINE
     if [ $VERSION_LINE -lt $VERSION_MATCH ] ; then
       #...If it is not then downloaded and unpacked
-      wget --no-check-certificate -P $DST $MKLURL -O $DST/$ARCHIVE_BASENAME
-      tar -xzf $DST/$ARCHIVE_BASENAME -C $DST
+      wget --no-check-certificate -P $MXNET_ROOT $MKLURL -O $MXNET_ROOT/$ARCHIVE_BASENAME
+      tar -xzf $MXNET_ROOT/$ARCHIVE_BASENAME -C $MXNET_ROOT
+      #echo $HOME_MKL
+      yes | cp -rf $MXNET_ROOT/$MKL_CONTENT_DIR/* $HOME_MKL
```

Comment:
I tested:
1. root user, put it in /usr/local
2. normal user, put it in ~
but for sudo , it is hard to do , becuase makefile need to get output from prepare_mkl.sh
if need sudo to input password, it will not work 
do you have seem any solution for this?